packages/api/src/command/medical/admin/redrive-dlq.ts

```diff
@@ -0,0 +1,93 @@
+import { SQS } from "aws-sdk";
+import { uniqBy } from "lodash";
+import BadRequestError from "../../../errors/bad-request";
+import { attributesToSend, getMessagesFromQueue, sqs } from "../../../external/aws/sqs";
+import { Config } from "../../../shared/config";
+import { uuidv4 } from "../../../shared/uuid-v7";
+
+const dlqUrl = Config.getSidechainFHIRConverterDQLURL();
+const regularUrl = Config.getSidechainFHIRConverterQueueURL();
+
+const maxNumberOfMessagesPerQuery = 10;
+const systemMaxNumberOfMessages = 100_000;
+
+/**
+ * ADMIN LOGIC - not to be used by other endpoints/services.
+ *
+ * Gets all messages from the Sidechain DLQ, unique, and sends them to the
+ * regular queue for re-processing.
+ */
+export async function redriveSidechainDLQ(maxNumberOfMessages: number): Promise<{
+  originalCount: number;
+  uniqueCount: number;
+}> {
+  if (!dlqUrl) throw new BadRequestError("Missing sidechain DLQ URL");
+  if (!regularUrl) throw new BadRequestError("Missing sidechain DLQ URL");
+  return redriveSQSUnique({
+    sourceQueueUrl: dlqUrl,
+    destinationQueueUrl: regularUrl,
+    maxNumberOfMessages,
+    contentComparator: (m: SQS.Message) => (m.Body ? JSON.parse(m.Body).s3FileName : ""),
+  });
+}
+
+// TODO consider exposing this to a shared library in core, along with Lambda's SQSUtils
+async function redriveSQSUnique({
+  sourceQueueUrl,
+  destinationQueueUrl,
+  maxNumberOfMessages,
+  contentComparator = defaultComparator,
+}: {
+  sourceQueueUrl: string;
+  destinationQueueUrl: string;
+  maxNumberOfMessages: number;
+  contentComparator?: ContentComparator;
+}): Promise<{
+  originalCount: number;
+  uniqueCount: number;
+}> {
+  const convertedMaxNumberOfMessages =
+    maxNumberOfMessages < 0 ? systemMaxNumberOfMessages : maxNumberOfMessages;
```

Comment:
nit: should this be `< 1`? not sure why we'd ever want to call this with `maxNumberOfMessages === 0`

Reply:
IMO zero is specific indicative of nothing, whereas `-1` is sometimes referred as invalid/no limit. I'd rather leave as is to better represent the user's intent. Who knows, we could execute it w/ zero to get an indication if there are messages there, for example.